python_modules/dagster-graphql/dagster_graphql/schema/external.py

```diff
@@ -273,160 +271,172 @@ class Meta:
 
     def __init__(
         self,
-        workspace_context: BaseWorkspaceRequestContext,
-        repository: ExternalRepository,
-        repository_location: CodeLocation,
+        handle: RepositoryHandle,
     ):
         # Warning! GrapheneAssetNode contains a GrapheneRepository. Any computation in this
         # __init__ will be done **once per asset**. Ensure that any expensive work is done
         # elsewhere or cached.
-        instance = workspace_context.instance
-        self._repository = check.inst_param(repository, "repository", ExternalRepository)
-        self._repository_location = check.inst_param(
-            repository_location, "repository_location", CodeLocation
-        )
-        check.inst_param(instance, "instance", DagsterInstance)
-        self._batch_loader = RepositoryScopedBatchLoader(instance, repository)
-        self._stale_status_loader = StaleStatusLoader(
-            instance=instance,
-            asset_graph=lambda: repository.asset_graph,
-            loading_context=workspace_context,
-        )
-        self._dynamic_partitions_loader = CachingDynamicPartitionsLoader(instance)
+        self._handle = handle
 
-        self._asset_graph_differ = None
-        # get_base_deployment_context is cached so there will only be one context per query
-        base_deployment_context = workspace_context.get_base_deployment_context()
-        if base_deployment_context is not None:
-            # then we are in a branch deployment
-            self._asset_graph_differ = AssetGraphDiffer.from_external_repositories(
-                code_location_name=self._repository_location.name,
-                repository_name=self._repository.name,
-                branch_workspace=workspace_context,
-                base_workspace=base_deployment_context,
+        self._batch_loader = None
+
+        super().__init__(name=handle.repository_name)
+
+    def get_repository(self, graphene_info: ResolveInfo) -> ExternalRepository:
+        return graphene_info.context.get_repository(self._handle.to_selector())
+
+    def get_batch_loader(self, graphene_info: ResolveInfo):
```

Comment:
any urge to do any of the async Loader business here?

Reply:
yea but not with much urgency